File: 1438-longest-continuous-subarray-with-absolute-diff-less-than-or-equal-to-limit/1438-longest-continuous-subarray-with-absolute-diff-less-than-or-equal-to-limit.py
class Solution:
    def longestSubarray(self, nums: List[int], limit: int) -> int:
        mn = deque()
        mx = deque()
        i = 0
        for a in nums:
            while(mn and mn[-1] > a): mn.pop()
            while(mx and mx[-1] < a): mx.pop()
            mn.append(a)
            mx.append(a)
            if(mx[0] - mn[0] > limit):
                if(mx[0] == nums[i]): mx.popleft()
                if(mn[0] == nums[i]): mn.popleft()
                i += 1
        return len(nums) - i
                
            

File: 1438-longest-continuous-subarray-with-absolute-diff-less-than-or-equal-to-limit/1438-longest-continuous-subarray-with-absolute-diff-less-than-or-equal-to-limit.go
func longestSubarray(nums []int, limit int) int {
    var maxQue = list.New()
    var minQue = list.New()
    var i = 0
    for _, num := range nums {
        for maxQue.Len() > 0 && maxQue.Back().Value.(int) < num {
            maxQue.Remove(maxQue.Back())
        }

        for minQue.Len() > 0 && minQue.Back().Value.(int) > num {
            minQue.Remove(minQue.Back())
        }
        
        maxQue.PushBack(num)
        minQue.PushBack(num)

        if (maxQue.Front().Value.(int) - minQue.Front().Value.(int)) > limit {
            if maxQue.Front().Value.(int) == nums[i] { maxQue.Remove(maxQue.Front()) }
            if minQue.Front().Value.(int) == nums[i] { minQue.Remove(minQue.Front()) }
            i += 1
        }
    }

    return len(nums) - i
}